features/steps/jobs_steps.go

```diff
@@ -452,3 +435,160 @@ func (f *JobsFeature) theJobsShouldBeOrderedByLast_updatedWithTheOldestFirst() e
+	var response models.Job
+
+	err := json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+
+	id = response.ID
+	err = f.PutNumberOfTasks(countStr)
+	if err != nil {
+		return err
+	}
+
+	return f.ErrorFeature.StepError()
+}
+
+// iCallPUTJobsNumber_of_tasksUsingAValidUUID is a feature step that can be defined for a specific JobsFeature.
+// It uses the parameters passed in to call PUT /jobs/{id}/number_of_tasks/{count}
+func (f *JobsFeature) iCallPUTJobsNumber_of_tasksUsingAValidUUID(idStr string, count int) error {
+
+	countStr := strconv.Itoa(count)
+	id = idStr
+
+	err := f.PutNumberOfTasks(countStr)
+	if err != nil {
+		return err
+	}
+
+	return f.ErrorFeature.StepError()
+}
+
+// iCallPUTJobsidnumber_of_tasksUsingTheGeneratedIdWithAnInvalidCount is a feature step that can be defined for a specific JobsFeature.
+// It gets the id from the response body, generated in the previous step, and then uses this to call PUT /jobs/{id}/number_of_tasks/{invalidCount}
+func (f *JobsFeature) iCallPUTJobsidnumber_of_tasksUsingTheGeneratedIdWithAnInvalidCount(invalidCount string) error {
+	f.responseBody, _ = ioutil.ReadAll(f.ApiFeature.HttpResponse.Body)
+	var response models.Job
+
+	err := json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+
+	id = response.ID
+
+	err = f.PutNumberOfTasks(invalidCount)
+	if err != nil {
+		return err
+	}
+
+	return f.ErrorFeature.StepError()
+}
+
+// iCallPUTJobsidnumber_of_tasksUsingTheGeneratedIdWithANegativeCount is a feature step that can be defined for a specific JobsFeature.
+// It gets the id from the response body, generated in the previous step, and then uses this to call PUT /jobs/{id}/number_of_tasks/{negativeCount}
+func (f *JobsFeature) iCallPUTJobsidnumber_of_tasksUsingTheGeneratedIdWithANegativeCount(negativeCount string) error {
+	f.responseBody, _ = ioutil.ReadAll(f.ApiFeature.HttpResponse.Body)
+	var response models.Job
+
+	err := json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+
+	id = response.ID
+
+	err = f.PutNumberOfTasks(negativeCount)
+	if err != nil {
+		return err
+	}
+
+	return f.ErrorFeature.StepError()
+}
+
+// theSearchReindexApiLosesItsConnectionToMongoDB is a feature step that can be defined for a specific JobsFeature.
+// It loses the connection to mongo DB by setting the mongo database to an invalid setting (in the Reset function).
+func (f *JobsFeature) theSearchReindexApiLosesItsConnectionToMongoDB() error {
+	f.Reset(true)
+	return nil
+}
+
+// GetJobByID is a utility function that is used for calling the GET /jobs/{id} endpoint.
+// It checks that the id string is a valid UUID before calling the endpoint.
+func (f *JobsFeature) GetJobByID(id string) error {
+	_, err := uuid.FromString(id)
+	if err != nil {
+		fmt.Println("Got uuid: " + id)
+		return err
+	}
+
+	// call GET /jobs/{id}
+	err = f.ApiFeature.IGet("/jobs/" + id)
+	if err != nil {
+		os.Exit(1)
```

Comment:
I think it would be helpful to have some kind of log to let us know why it's failing here

Reply:
Actually why is it exiting rather than returning the error?

Reply:
I think there's a historical reason for that but I'm not sure what it was. In refactoring this code copied this line from when the tests had first started being written.